documents/trail/ecs-mini-program.yml

```diff
@@ -0,0 +1,353 @@
+      zh-cn: ECS实例ID
+    AssociationProperty: 'ALIYUN::ECS::Instance::InstanceId'
+  EcsRamRoleName:
+    Type: String
+    Label:
+      en: RAM Role Name
+      zh-cn: RAM角色名称
+Resources:
+  DS_Instances:
+    Type: 'DATASOURCE::ECS::Instances'
+    Properties:
+      InstanceIds:
+        - Ref: InstanceId
+  WaitConditionHandle:
+    Type: 'ALIYUN::ROS::WaitConditionHandle'
+    Properties: {}
+  WaitCondition:
+    Type: 'ALIYUN::ROS::WaitCondition'
+    Properties:
+      Handle:
+        Ref: WaitConditionHandle
+      Timeout: 3720
+      Count: 1
+  EcsRamRole:
+    Type: 'ALIYUN::RAM::Role'
+    Properties:
+      RoleName:
+        Ref: EcsRamRoleName
+      Description: 搭建支付宝小程序ECS实例角色
+      AssumeRolePolicyDocument:
+        Version: '1'
+        Statement:
+          - Action: 'sts:AssumeRole'
+            Effect: Allow
+            Principal:
+              Service:
+                - ecs.aliyuncs.com
+      Policies:
+        - PolicyName:
+            'Fn::Sub':
+              - '${RamRoleName}Policy'
+              - RamRoleName:
+                  Ref: EcsRamRoleName
+          Description: 搭建支付宝小程序ECS实例角色权限策略
+          PolicyDocument:
+            Version: '1'
+            Statement:
+              - Action:
+                  - 'ecs:DescribeInstances'
+                  - 'ecs:DescribeInstanceStatus'
+                Effect: Allow
+                Resource:
+                  - '*'
+  EcsRamRoleAttachment:
+    Type: 'ALIYUN::ECS::RamRoleAttachment'
+    Properties:
+      InstanceIds:
+        - Ref: InstanceId
+      RamRoleName:
+        Ref: EcsRamRoleName
+    DependsOn: EcsRamRole
+  SecurityGroupIngress_80:
+    Type: 'ALIYUN::ECS::SecurityGroupIngress'
+    Properties:
+      SecurityGroupId:
+        'Fn::Jq':
+          - First
+          - '.[0].SecurityGroupIds[0]'
+          - 'Fn::GetAtt':
+              - DS_Instances
+              - Instances
+      SourceCidrIp: 0.0.0.0/0
+      IpProtocol: tcp
+      NicType: intranet
+      PortRange: 80/80
+  InstallWebServer:
+    Type: 'ALIYUN::ECS::RunCommand'
+    Properties:
+      InstanceIds:
+        - Ref: InstanceId
+      Type: RunShellScript
+      Sync: true
+      Timeout: 3600
+      CommandContent:
+        'Fn::Sub': >-
+          #!/bin/bash
+
+
+          if [ ! -f .ros.provision ]; then
+            echo "Name: 搭建支付宝小程序" > .ros.provision
+          fi
+
+
+
+          name=$(grep "^Name:" .ros.provision | awk -F':' '{print $2}' | tr -d
+          '[:space:]')
+
+
+          if [[ "$name" != "搭建支付宝小程序" ]]; then
+            echo "当前实例已使用过\"$name\"教程的一键配置，不能再使用本教程的一键配置"
+            ${WaitConditionHandle.CurlCli} --data-binary "{\"status\": \"FAILURE\", \"reason\": \"The current instance has already applied the configuration of the \\\"$name\\\" tutorial, and the configuration of this tutorial can no longer be applied.\"}"
+            exit 0
+          fi
+
+
+          if ! grep -q "^Step1: Install Python$" .ros.provision; then
```

Comment:
The judgment here seems to have no effect, because the content is not written to the  `.ros.provision` file.